tests/Schema/ComparatorTest.php

```diff
@@ -1337,4 +1337,37 @@ public function testNoOrphanedForeignKeyIfReferencingTableIsDropped(): void
 
         self::assertEmpty($diff->orphanedForeignKeys);
     }
+
+    public function testDiffTextTypeValidChange(): void
+    {
+        $column1 = new Column('col1', Type::getType('text'));
+        $column1->setLength(65536);
+
+        $column2 = new Column('col2', Type::getType('text'));
+        $column2->setLength(16777216);
+
+        self::assertTrue($this->comparator->diffTextType($column1, $column2));
```

Comment:
The newly added logic and assertions are wrong since the actual comparison results will depend on the target database platform. See https://github.com/doctrine/dbal/pull/4746 for more details.